### ISSUE TITLE
Integrate BreakID

### ISSUE THREAD

**Issue by a1880** [msoos/cryptominisat#639]
Hi Mate,

I have built Bosphorus/Cryptominisat under Ubuntu 20.04.
As a prerequisite (?), I also installed BreakID.
However, I could not make the BreakID location known to cmake.
Therefore, BreakID is not recognized as available package.

It would be nice, if the installation instruction could cover this.

What features are missing without BreakID?

Greetings,

Axel

**Comment by msoos** [comment]
Hi!

Happy New Year!

I worked like 2 months on BreakID to be integrated. I am happy to give you
access in case you promise you will not publish it, however, I am reluctant
to make it public -- I've had sad moments when people abused my trust and
decided to publish something I have worked on. I hope this is
understandable. Let me know if you need access,

Mate

**Comment by a1880** [comment]
Hello Mate,

instant reply, as always. I hope, you are well.

I did some experiments with BreakID and symmetry breaking. My matrix multiplication ("MM") problem is highly symmetric.
But my results were rather disappointing. The CNF grew a lot, and the solving times went up.

Now, I go my first steps with Bosphorus. MM can be written as ANF. 
I noticed that the process ANF --> solution is much slower than ANF --> CNF --> solution.
Reason is probably that I can adjust the SAT solver settings, if I call the SAT solver separately.

I don't plan to publish anything. If I should ever find a matrix multiplication algorithm with less than 23 products for 3x3 matrices, a paper would follow. But this appears less and less likely after many attempts.

Cheers,

Axel
